### ISSUE TITLE
pattern matched unions are formatted badly

### ISSUE THREAD

**Issue by 7sharp9** [fsprojects/fantomas#283]
Using `sourceTransformer.fs` from this repo 

```fsharp
let rec multiline synExpr = 
    match synExpr with
    | ConstExpr _
    | NullExpr
    | OptVar _
    | SequentialSimple _ ->
        false

    | ObjExpr _
    | While _
    | For _
    | ForEach _
    | TryWith _
    | TryFinally _
    | Sequentials _
    | IfThenElse _ ->
        true

    | Paren e
    | SingleExpr(_, e)
    | TypedExpr(_, e, _)
    | CompExpr(_, e)
    | ArrayOrListOfSeqExpr(_, e)
    | DesugaredLambda(_, e)
    | Lambda(e, _)
    | TypeApp(e, _)
    | LongIdentSet(_, e)
    | DotGet(e, _)
    | TraitCall(_, _, e) ->
        multiline e
...
```

is formatted into this:

```fsharp
let rec multiline synExpr =
    match synExpr with
    | ConstExpr _ | NullExpr | OptVar _ | SequentialSimple _ -> false
    | ObjExpr _ | While _ | For _ | ForEach _ | TryWith _ | TryFinally _ | Sequentials _ | IfThenElse _ -> true
    | Paren e | SingleExpr(_, e) | TypedExpr(_, e, _) | CompExpr(_, e) | ArrayOrListOfSeqExpr(_, e) | DesugaredLambda(_, 
                                                                                                                      e) | Lambda(e, 
                                                                                                                                  _) | TypeApp(e, 
                                                                                                                                               _) | LongIdentSet(_, 
                                                                                                                                                                 e) | DotGet(e, 
                                                                                                                                                                             _) | TraitCall(_, 
                                                                                                                                                                                            _, 
                                                                                                                                                                                            e) -> 
        multiline e
```
Another piece of code in this file:

```fsharp
let (|OneLinerBinding|MultilineBinding|) b =
    match b with
    | LetBinding([], PreXmlDoc [||], _, _, _, _, OneLinerExpr _)
    | DoBinding([], PreXmlDoc [||], OneLinerExpr _)
    | MemberBinding([], PreXmlDoc [||], _, _, _, _, OneLinerExpr _)
    | PropertyBinding([], PreXmlDoc [||], _, _, _, _, OneLinerExpr _) 
    | ExplicitCtor([], PreXmlDoc [||], _, _, OneLinerExpr _, _) -> 
        OneLinerBinding b

    | _ -> MultilineBinding b
```

Is formatted to this:

```fsharp
let (|OneLinerMemberDefn|MultilineMemberDefn|) md =
    match md with
    | MDOpen _ | MDInherit _ | MDValField _ | MDImplicitCtor _ | MDInterface(_, None) | MDAbstractSlot([], 
                                                                                                       PreXmlDoc [||], _, 
                                                                                                       _, _, _, _, _) | MDImplicitInherit(_, 
                                                                                                                                          OneLinerExpr _, 
                                                                                                                                          _) | MDMember(OneLinerBinding _) | MDAutoProperty([], 
                                                                                                                                                                                            PreXmlDoc [||], 
                                                                                                                                                                                            _, 
                                                                                                                                                                                            _, 
                                                                                                                                                                                            OneLinerExpr _, 
                                                                                                                                                                                            _, 
                                                                                                                                                                                            _, 
                                                                                                                                                                                            _, 
                                                                                                                                                                                            _) | MDLetBindings(_, 
                                                                                                                                                                                                               _, 
                                                                                                                                                                                                               [ OneLinerBinding _ ]) -> 
        OneLinerMemberDefn md
    | _ -> MultilineMemberDefn md
```

There are multiple issues with indentation and union filed components being moved down a line.

**Comment by 7sharp9** [comment]
I suppose but thats just disabling the issue

**Comment by nojaf** [comment]
Would it be ok to state that every case inside a pattern match should be on a newline?
[Style guide](https://docs.microsoft.com/en-us/dotnet/fsharp/style-guide/formatting#pattern-matching-constructs) says `If the expression is short, you can consider using a single line if each subexpression is also simple.`. 

However to me it doesn't seem like the end of the world if we go with an opinionated approach and go with a line anyway. Or does anyone have a good heuristic in mind?

**Comment by jindraivanek** [comment]
Fixed by #391.
